### ISSUE TITLE
Default SameSite to Lax?

### ISSUE THREAD

**Issue by rowan-m** [WICG/cookie-store#182]
I saw this was initially covered in #36, but I think it's worth revisiting now.

https://github.com/WICG/cookie-store/blob/9a100293eb01c0828fea16d266ea6d410ef6934f/index.bs#L497

The default `SameSite` value is `Strict`. We've made the default treatment for cookies elsewhere as `Lax`. This feels too me as if developers will need to add boilerplate to set `Lax` on most of their cookies, along with this generally feeling a bit inconsistent and therefore confusing.

I get the motivation for good defaults, but disagree that `Strict` is good here.

**Comment by annevk** [comment]
I'm not sure we should equate the default for existing cookies with the default for new cookies. Why would developers mostly use Lax?

**Comment by rowan-m** [comment]
I'll retract "most" as that's not backed up with evidence, just personally dealing with a lot of developer confusion around `SameSite` as a concept at all 😅

Rather, it feels confusing that these yield different results:

```js
cookieStore.set('opted_out', '1');
document.cookie = 'opted_out=1';
```

However, after playing around a bit I'm realising that `Lax` is primarily beneficial if you want the cookie sent in the **header** on navigation, client-side `SameSite=Strict` cookies are actually available within the page on navigation. So, maybe the impact on apps that are primarily using cookies within JavaScript is minimal, `Strict` does seem like a good default there. It's only instances where a developer is setting a cookie from JavaScript that they would want sent to their server on initial navigation.

I do like the idea of encouraging developers to set explicit defaults, so I'm not overly attached to this, but at the minimum I think there should be a clear "Differences from `document.cookie`" or "Migration guide" section.

**Comment by aamuley** [comment]
Hey just wanted to follow up on this since there is more adoption of the API now and the cross-browser implementations are aligned on using `Strict` by default. 

While using `Lax` could help make the CookieStore API more interoperable with `document.cookie` or HTTP cookie defaults, 
- Making it truly interoperable would suggest `lax-allowing-unsafe` should be the default which is unideal
- Changing the default from `Strict` to `Lax` could be a security sensitive change 

I think Chromium is okay with sticking to Strict by default unless @annevk or @bakulf have any reservations or preferences? 

(and @rowan-m we can definitely call this out in the spec if that would abate some developer confusion?)

**Comment by annevk** [comment]
Agreed that this shouldn't change at this point. Calling it out sounds reasonable, but we also haven't gotten any further reports about this or thumbs up on OP so unclear how necessary that is.

**Comment by DCtheTall** [comment]
Since there seems to be consensus around keeping `Strict` as the default, I think can should close this. The current draft of the standard already instructs browsers to keep using `Strict`.
